infrasim/model/elements/pci_imc.py

```diff
@@ -37,13 +37,17 @@ def init(self):
         self.id = self.__imc_info["id"]
         self.bus = self.__imc_info["bus"]
         self.addr = self.__imc_info.get("addr")
+        self.imc_slot_topo = self.__imc_info.get("imc_slot_topo")
+        self.imc_cpu_index = self.__imc_info.get("imc_cpu_index")
```

Comment:
Are these two arguments optional or mandatory?
If they are optional, you should check if they have values in handle_parms before inserting them into args.

Reply:
Yes, they are mandatory for warnado platform.  
Precheck function code updated.